### PR TITLE
Be more lenient when parsing json from RapidAntigen QRCodes (DEV)

### PR DESCRIPTION
**Pull request by d4rken**
+ We allow empty strings and treat them as null.
+ some extra unit tests

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt [hunk 1]
```diff
@@ -78,15 +50,57 @@ class RapidAntigenQrCodeExtractor @Inject constructor() : QrCodeExtractor<Corona
...
+                return rawHash
+            }
+
+        val createdAt: Instant
+            get() {
+                if (rawTimestamp == null || rawTimestamp <= 0) throw InvalidQRCodeException("Timestamp is invalid")
+                return Instant.ofEpochSecond(rawTimestamp)
+            }
+
+        val firstName: String?
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt:76]: what about blank Strings?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt [hunk 2]
```diff
@@ -78,15 +50,57 @@ class RapidAntigenQrCodeExtractor @Inject constructor() : QrCodeExtractor<Corona
...
+            get() {
+                if (rawFirstName.isNullOrEmpty()) return null
+                return rawFirstName
+            }
+
+        val lastName: String?
```
Comment (BMItter) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt:82]: what about blank Strings?...

**Review by BMItr — APPROVED** [review]

**Review by chiljamgossow — APPROVED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt (hunk 2 above)
Comment (harambasicluka) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt:82]: @d4rken I think blank strings can be used for the birthdays, not sure about the last name.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/qrcode/RapidAntigenQrCodeExtractor.kt:76]: I don't know. I guess `‎‎‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎    ,  ‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎‏‏‎ ‎    ` should not be excluded from taking RA tests? :shrug: /cc @mlenkeit

**Pull request merged**